### PR TITLE
fix: #6680, Tooltip: Escape key not dismissing tooltip (WCAG 2.1 1.4.13)

### PR DESCRIPTION
**Pull request by akshayaqburst**
### Defect Fixes
fix: #6680, Tooltip: Escape key not dismissing tooltip (WCAG 2.1 1.4.13)

### CODE REVIEW

**Comment by akshayaqburst** [comment]
Anyway, we have to specify closeOnEscape in the tooltipOptions prop, which is already specified in the documentation.

                <InputText value={value} onChange={(e) => setValue(e.target.value)}  tooltip="tooltip content" tooltipOptions={{closeOnEscape: true}} />

File: components/lib/hooks/useGlobalOnEscapeKey.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { DomHandler } from '../utils/Utils';
  * Priorities of different components (bigger number handled first)
  */
 export const ESC_KEY_HANDLING_PRIORITIES = {
+    TOOLTIP: 10,
```
Comment (melloware) [inline components/lib/hooks/useGlobalOnEscapeKey.js:8]: I wonder if tooltip should be 1200?  I would think tooltips should always be on top of everything else?

File: components/lib/tooltip/Tooltip.js [hunk 2]
```diff
@@ -32,6 +32,13 @@ export const Tooltip = React.memo(
         const { ptm, cx, sx, isUnstyled } = TooltipBase.setMetaData(metaData);
 
         useHandleStyle(TooltipBase.css.styles, isUnstyled, { name: 'tooltip' });
+        useGlobalOnEscapeKey({
+            callback: () => {
+                hide();
+            },
+            when: props.closeOnEscape,
+            priority: [ESC_KEY_HANDLING_PRIORITIES.TOOLTIP, 0]
```
Comment (melloware) [inline components/lib/tooltip/Tooltip.js:40]: I think closeOnEscpae might be missing from the tooltip options in the typescript definition

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/tooltip/Tooltip.js (hunk 2 above)
Comment (akshayaqburst) [reply components/lib/tooltip/Tooltip.js:40]: Added. @melloware

**Review by melloware — APPROVED** [review]

**Pull request merged**